### PR TITLE
Add lexicon.txt to demos, libraries, and platform.

### PR DESCRIPTION
**Pull request by sarenameas**
*Description of changes:*
- Add lexicon.txt to demos, platform, libraries/standard/http, libraries/standard/json, and libraries/standard/mqtt.
- Fix spelling mistakes where they were found.
- Update script to ignore the third_party folder.
- Add README.md on how to create a new lexicon.txt.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/lexicon.txt [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+createcleansession
+currentstate
+defragmenting
+deserialization
+deserialize
+deserializeack
+deserialized
+deserializepublish
+deserializing
+didn
```
Comment (sarenameas) [inline libraries/standard/mqtt/lexicon.txt:35]: The part before the apostrophe in contractions like *hasn't*, *doesn't* and *didn't* are flagged by the script and need to be part of the lexicon.

Comment (abhidixi11) [reply libraries/standard/mqtt/lexicon.txt:35]: Was this the problem for v4_beta? How did we handle it there ?

File: demos/lexicon.txt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+psessionpresent
+ptransportinterface
+puback
+pubcomp
+publishpacketsent
+pubrec
+pubrel
+qos
+resending
+resp
```
Comment (aggarw13) [inline demos/lexicon.txt:84]: Name of a parameter or member?

File: libraries/standard/http/lexicon.txt [hunk 3]
```diff
@@ -0,0 +1,144 @@
+absolutevalue
+addrangeheader
+addrangerequest
+aggregator
+api
+ascii
+belown
```
Comment (aggarw13) [inline libraries/standard/http/lexicon.txt:7]: Shouldn't be supported?

File: libraries/standard/http/lexicon.txt [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+httpparserxxxxcallback
+httpparserxxxxcallbacks
+httprequestinfo
+https
+ietf
+ifndef
+inc
+init
+isheaderresponse
+isheadresponse
```
Comment (aggarw13) [inline libraries/standard/http/lexicon.txt:54]: Typo?

File: demos/lexicon.txt (hunk 2 above)
Comment (sarenameas) [reply demos/lexicon.txt:84]: yes

File: libraries/standard/http/lexicon.txt (hunk 3 above)
Comment (sarenameas) [reply libraries/standard/http/lexicon.txt:7]: This was fixed but missed from deleting thanks for catching

File: libraries/standard/http/lexicon.txt (hunk 4 above)
Comment (sarenameas) [reply libraries/standard/http/lexicon.txt:54]: No

File: libraries/standard/mqtt/lexicon.txt (hunk 1 above)
Comment (aggarw13) [reply libraries/standard/mqtt/lexicon.txt:35]: This was the same behavior in `v4_beta` and we used the same solution in the lexicon files there.

File: libraries/standard/mqtt/lexicon.txt [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+sublicense
+subscriptioncount
+subscriptiontype
+tcp
+tcpsocket
+testcase
+timeoutms
+tls
+uint
+un
```
Comment (aggarw13) [inline libraries/standard/mqtt/lexicon.txt:217]: Does this represent a word of the kind "_un_-<word>"?

File: platform/lexicon.txt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+https
+ifndef
+implemenation
+inc
+iot
+logpath
+malloc
+maxattempts
+messagelevel
+mfln
```
Comment (aggarw13) [inline platform/lexicon.txt:33]: ?

Comment (sarenameas) [reply platform/lexicon.txt:33]: that is a protocol, if you are confused about a word, please do a search for it in your code.

File: libraries/standard/mqtt/lexicon.txt (hunk 5 above)
Comment (sarenameas) [reply libraries/standard/mqtt/lexicon.txt:217]: Yes

**Pull request merged**